cfme/common/provider.py

```diff
@@ -101,6 +103,46 @@ def deployment_helper(self, deploy_args):
         """ Used in utils.virtual_machines and usually overidden"""
         return {}
 
+    @property
+    def default_endpoint(self):
+        try:
+            return self.endpoints['default']
+        except (KeyError, AttributeError):
+            return None
+
+    @property
+    def hostname(self):
+        if self.default_endpoint:
+            return self.default_endpoint.hostname
+        else:
+            return None
```

Comment:
I was definitely sure that getattr(None, "something", None) would cause exception. I was wrong.